# planner_interface configuration
cartesian_velocity: 0.05 # [m/s]
welding_velocity: 300 # [cm/min], see velocity_convert_ in welding.h
tool_coupling_velocity: 0.01 # [m/s]
scanning_velocity: 0.2 # [m/s]
slag_removal_velocity: 0.006666 # [m/s]
via_velocity: 1 # [m/s]
cartesian_rot_velocity: 1.56 # [rad/s]
away_max_joint_velocity: 0.2 #
via_max_joint_velocity: 1 #
# away_max_joint_velocity: -1 # [rad/s] if <=0, default will be used
# via_max_joint_velocity: -1 # [rad/s] if <=0, default will be used
max_acceleration_scaling: 0.1 # [rad/sÂ²]
step_size: 0.001
num_planning_attempts: 1
goal_joint_tolerance: 0.00001 # max over axes [rad]
goal_position_tolerance: 0.001 # [m]
goal_orientation_tolerance: 0.5 # [rad]
linear_planner_id: "LIN"
circular_planner_id: "CIRC"
curve_planner_id: "SPLINE"
joint_space_planner_id: "PTP"
constrained_planner_id: "RRTstar"
planner_id_property_name: "planner"

planning_time_free_space: 1
planning_time_constrained: 5
planning_time_collisions: 90 # not implemented yet
# Total planning attempts
max_solutions: 10


# Planning group and link names
arm_group_name: "ur_manipulator"
moveit_ee_name: "endeffector" #"flange"
welding_group_name: "welding_endeffector" # "welding_arm"
welding_tcp_group: "welding_gun" # "welding_tcp"
welding_tcp_frame: "tcp_welding_gun_link" # "welding_tcp"
slag_group_name: "slag_endeffector" #
slag_tcp_frame: "slag_brush" #

# Offsets for specific sub-tasks
## Welding
offset_welding_approach_z: 100 #[mm]
offset_welding_z: 5 #22 #[mm]
offset_welding_y: -9 #-5 #[mm]
offset_welding_x: 2.5 #2 #-2 #3 #[mm]
offset_welding_angle: 0 #3.5 #[deg]
offset_welding_speed: 0 #[cm/min]
## Scanning
offset_scanning_approach_z: 100 #[mm]
offset_scanning_x: 20 #0.03 #[mm] positive upwards
offset_scanning_y: 80 #0.03 #[mm] positive right
offset_scanning_z: 70 #0.08 #[mm] positive stickout
offset_scanning_a: 0 #[deg]
offset_scanning_b: 0 #[deg]
offset_scanning_c: 0 #[deg]

# Poses
## Robot configuration in joint space
arm_home_pose: "ready"
arm_interim_home_pose: "scanning_to_home"

# Scene frames
world_frame: "world"
workpiece_frame: "workpiece"
manufacturing_to_task_frame: "ManufacturingToTaskFrame"
